update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,351 @@
+import no.unit.nva.model.DoiRequest;
+import no.unit.nva.model.DoiRequestStatus;
+import no.unit.nva.model.Publication;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.exceptions.commonexceptions.NotFoundException;
+import nva.commons.handlers.GatewayResponse;
+import nva.commons.utils.Environment;
+import nva.commons.utils.log.LogUtils;
+import nva.commons.utils.log.TestAppender;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+public class ApiUpdateDoiRequestHandlerTest extends DoiRequestsDynamoDBLocal {
+
+    public static final String INVALID_PUBLICATION_ID = "InvalidPublicationId";
+    public static final String NULL_STRING_REPRESENTATION = "null";
+    public static final String FAKE_ENV_SCHEMA_AND_HOST = FAKE_API_SCHEME_ENV
+        + "://"
+        + FAKE_API_HOST_ENV
+        + "/publication/";
+    private static final String INVALID_USERNAME = "invalidUsername";
+    private static final String USERNAME_NOT_IMPORTANT = INVALID_USERNAME;
+    private final Environment environment;
+    private final String publicationsTableName;
+    private final Context context;
+    private final Instant mockNow = Instant.now();
+    private final Instant mockOneHourBefore = mockNow.minus(1, ChronoUnit.HOURS);
+    private DynamoDBDoiRequestsService doiRequestsService;
+    private UpdateDoiRequestHandler handler;
+
+    public ApiUpdateDoiRequestHandlerTest() {
+        environment = mockEnvironment();
+        publicationsTableName = environment.readEnv(ServiceConstants.PUBLICATIONS_TABLE_NAME_ENV_VARIABLE);
+        context = mock(Context.class);
+    }
+
+    @BeforeEach
+    public void init() {
+        initializeDatabase();
+        Clock mockClock = getFixedClockWithDefaultTimeZone(mockNow);
+        doiRequestsService = new DynamoDBDoiRequestsService(client, objectMapper, environment, mockClock);
+        handler = new UpdateDoiRequestHandler(environment, doiRequestsService);
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenPublicationIdIsInvalid() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
+        updateDoiRequest.setPublicationId(INVALID_PUBLICATION_ID);
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(details.getDetail(), containsString(INVALID_PUBLICATION_ID_ERROR));
+        assertThat(details.getDetail(), containsString(updateDoiRequest.getPublicationId()));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenNoPublicationId() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
```

Comment:
Inline

Reply:
This is used in multiple tests, and @axthosarouris don't want code duplication. Keeping it as is.